plugins/modules/wafv2_ip_set.py

```diff
@@ -0,0 +1,325 @@
+      type: str
+    description:
+      description:
+        - Description of the IP set.
+      required: false
+      type: str
+    scope:
+      description:
+        - Specifies whether this is for an AWS CloudFront distribution or for a regional application,
+          such as API Gateway or Application LoadBalancer.
+      choices: ["CLOUDFRONT","REGIONAL"]
+      required: true
+      type: str
+    ip_address_version:
+      description:
+        - Specifies whether this is an IPv4 or an IPv6 IP set.
+        - Required when I(state=present).
+      choices: ["IPV4","IPV6"]
+      type: str
+    addresses:
+      description:
+        - Contains an array of strings that specify one or more IP addresses or blocks of IP addresses in
+          Classless Inter-Domain Routing (CIDR) notation.
+        - Required when I(state=present).
+        - When I(state=absent) and I(addresses) is defined, only the given IP addresses will be removed
+          from the IP set. The entire IP set itself will stay present.
+      type: list
+      elements: str
+    tags:
+      description:
+        - Key value pairs to associate with the resource.
+        - Currently tags are not visible. Nor in the web ui, nor via cli and nor in boto3.
+      required: false
+      type: dict
+    purge_addresses:
+      description:
+        - When set to C(no), keep the existing addresses in place. Will modify and add, but will not delete.
+      default: yes
+      type: bool
+
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = '''
+- name: test ip set
+  wafv2_ip_set:
+    name: test02
+    state: present
+    description: hallo eins
+    scope: REGIONAL
+    ip_address_version: IPV4
+    addresses:
+      - 8.8.8.8/32
+      - 8.8.4.4/32
+    tags:
+      A: B
+      C: D
+'''
+
+RETURN = """
+addresses:
+  description: Current addresses of the ip set
+  sample:
+    - 8.8.8.8/32
+    - 8.8.4.4/32
+  returned: Always, as long as the ip set exists
+  type: list
+arn:
+  description: IP set arn
+  sample: "arn:aws:wafv2:eu-central-1:11111111:regional/ipset/test02/4b007330-2934-4dc5-af24-82dcb3aeb127"
+  type: str
+  returned: Always, as long as the ip set exists
+description:
+  description: Description of the ip set
+  sample: Some IP set description
+  returned: Always, as long as the ip set exists
+  type: str
+ip_address_version:
+  description: IP version of the ip set
+  sample: IPV4
+  type: str
+  returned: Always, as long as the ip set exists
+name:
+  description: IP set name
+  sample: test02
+  returned: Always, as long as the ip set exists
+  type: str
+"""
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule, is_boto3_error_code, get_boto3_client_method_parameters
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict, ansible_dict_to_boto3_tag_list
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError, WaiterError
```

Comment:
Could you please wrap your boto client calls in try/except blocks that handle both ClientError and BotoCoreError, and exits with module.fail_json_aws? (that method will handle to_native on the exception message for you)

I don't see any use of waiters so we shouldn't need to import WaiterError